tests/Collector/EventListener/SchemaFilterListenerTest.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Bundle\MigrationsBundle\Tests\Collector\EventListener;
+
+use Doctrine\Bundle\MigrationsBundle\EventListener\SchemaFilterListener;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\Migrations\Tools\Console\Command\DoctrineCommand;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Console\Event\ConsoleCommandEvent;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Output\NullOutput;
+
+class SchemaFilterListenerTest extends TestCase
+{
+    public function testItFiltersOutMigrationMetadataTableByDefault(): void
+    {
+        $listener = new SchemaFilterListener('doctrine_migration_versions');
+
+        self::assertFalse($listener(new Table('doctrine_migration_versions')));
+        self::assertTrue($listener(new Table('some_other_table')));
+    }
+
+    public function testItFiltersNothingWhenDisabled(): void
+    {
+        $listener = new SchemaFilterListener('doctrine_migration_versions');
+        $listener->disable();
+
+        self::assertTrue($listener(new Table('doctrine_migration_versions')));
+        self::assertTrue($listener(new Table('some_other_table')));
+    }
```

Comment:
I don't understand this use case. When do you need to disable it in userland by calling the public disable-method?

Reply:
:thinking: I suspect I might have written that one before writing the next use case. I'll remove it and make the method private, if it's needed in userland, people can make it public later.